panel/widgets/input.py

```diff
@@ -98,7 +89,41 @@ def from_param(cls, parameter: param.Parameter, onkeyup=False, **params) -> View
         return super().from_param(parameter, **params)
 
 
-class PasswordInput(TextInput):
+class TextInput(_TextInputBase):
+
+    """
+    The `TextInput` widget allows entering any string using a text input box.
+
+    Reference: https://panel.holoviz.org/reference/widgets/TextInput.html
+
+    :Example:
+
+    >>> TextInput(name='Name', placeholder='Enter your name here ...')
+    """
+
+    enter_pressed = param.Event(doc="""
+        Event when the enter key has been pressed.""")
+
+    _widget_type: ClassVar[Type[Model]] = _BkTextInput
+
+    _rename = {'enter_pressed': None}
+
+    def _get_model(
+        self, doc: Document, root: Optional[Model] = None,
+        parent: Optional[Model] = None, comm: Optional[Comm] = None
+    ) -> Model:
+        model = super()._get_model(doc, root, parent, comm)
+        self._register_events('enter-pressed', model=model, doc=doc, comm=comm)
+        return model
+
+    def _process_event(self, event) -> None:
+        if event.event_name == 'enter-pressed':
+            self.value = event.value_input
+            self.value_input = event.value_input
```

Comment:
Thinking about using `param.discard_events` for these two.

Reply:
Don't think that'll work since it'll suppress the subsequent change event as well since at that point the value won't changed. Events generally are processed before any value change and if we set it here then that rule is violated but I get that the values have to be set before the event.